convertMQTTTopicToRegex.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * This is the shared prefix name specified by the MQTT spec. When you
+ * subscribe to a topic, if you follow the format $share/group_name/topic_name,
+ * the messages will be dispersed to ONE consumer subscribed to that topic_name,
+ * and the incoming topic name will be "topic_name".
+ * @type {string}
+ */
+const sharedPrefix = '$share/';
+
+function convertMQTTTopicToRegex(topic) {
+    let strippedTopic = topic;
+
+    if (strippedTopic.startsWith(sharedPrefix)) {
+        let mangledTopic = strippedTopic.slice(sharedPrefix.length);
+        // do we have a group? It'll be shown here.
+        const indexOfTopic = mangledTopic.indexOf('/');
+
+        if (indexOfTopic > 0) {
+            mangledTopic = mangledTopic.slice(indexOfTopic + 1);
+        }
+
+        strippedTopic = mangledTopic;
+    }
+
+    const convertedTopic = strippedTopic
+        .replace('+', '[^/]+')
+        .replace('#', '/.+');
```

Comment:
It would be great to have some unit tests for this function. 
I am not sure that this case is covered:
subscription: /aa/#
regex: /aa//.+ (we have two slashes)
pub: /aa/bb (probably will not match)

Reply:
Need to check this, too, but leading slashes are not recommended for topic names. My thought is the regex handles this properly, but will gladly add tests.